### ISSUE TITLE
Get-Notebooks failing since v2.1.3771

### ISSUE THREAD

**Issue by dsuarez993** [DataThirstLtd/azure.databricks.cicd.tools#140]
At my organization we ran into an issue since last change on v2.1.3771.

Randomly (9 out of 10 times), when we were running our CI/CD pipeline, we were getting an error while calling Export-DatabricksFolder. The error was pointing to Get-Notebooks.

After digging into the repo history, we found out you updated this specific script. Moving back to v2.1.2915 solved the issue for us.
 
Sorry, because I can't give you more hints since I haven't been able to replicate this error on my machine. 

We are making the call as follows:
`
Export-DatabricksFolder -BearerToken $BearerToken -Region $Region -ExportPath "/$($WORKSPACEPATH)" -LocalOutputPath "$($REL_FOLDER_OUTPUT)\$($WORKSPACEPATH)"
`

The error:
`##[error]Get-Notebooks : Oh dear one of the jobs has failed. Check the details of the jobs above.
At C:\Program Files\WindowsPowerShell\Modules\azure.databricks.cicd.tools\2.1.3786\Private\Get-Notebooks.ps1:44 char:13
            Get-Notebooks $SubfolderContents ($Object.path + "/") $Lo ...
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Get-Notebooks`

Any idea? Could anyone look into this?

**Comment by RichieBzzzt** [comment]
Hello! 

Looks like one of the threadjobs that gets the notebook content and then creates or updates the local file. Looking through the code I see that it should write-host the error above this error message.  https://github.com/DataThirstLtd/azure.databricks.cicd.tools/blob/b51cd450ed9b562d059e8a9215aca93c3bd66f81/Private/Get-Notebooks.ps1#L80-L91

Are you able to share any more of the logs form the failed pipelines? And what sort of notebooks are you using (ie .py, html etc etc)

**Comment by dsuarez993** [comment]
I've tried again with the latest release 2.1.3834, but still same error.

I did some further investigation, and I found out that one of the Notebooks was empty.
After removing the file, the Export-Notebooks command worked again.
In order to be sure, I wanted to replicate the error by creating another empty notebook, and yes, the error appeared again.

So I decided to add -Verbose to the command to get detailed traces and yes! That was it! As you can see in the screenshot, it fails to export the "empty" notebook (which is a .py notebook with an empty cell).

![ErrorExportNotebook](https://user-images.githubusercontent.com/31488342/96137598-1d898a80-0efd-11eb-838b-303748342543.png)

The empty notebook looks like this:

![EmptyNotebook](https://user-images.githubusercontent.com/31488342/96139246-f469f980-0efe-11eb-952c-e062e5952bea.png)


Once I've spotted the error, I tried to replicate it on my laptop. So I ran the same command with the latest release 2.1.3834 version, but it was impossible to get the error there, everything worked fine (even with the empty notebook in the Databricks Workspace).

I don't know what it can be, maybe the PowerShell version we use in the Task?

![AgentPowerShellVersion](https://user-images.githubusercontent.com/31488342/96139042-b8369900-0efe-11eb-974c-81884380a862.png)

Any idea??

Now that I know this issue is here I know where to look at, and how to prevent it, but I think it would be nice to see why this is actually happening

Thanks!

**Comment by RichieBzzzt** [comment]
Hiya, 

I've managed to find the offending line. 

It was this ```            $Response = Get-Content $tempLocalExportPath -Encoding UTF8 ```

But the below pending change should resolve it. 

https://github.com/RichieBzzzt/azure.databricks.cicd.tools/blob/13836855ecd99e8401cb110aeec6c01b15486266/Private/Get-Notebooks.ps1#L12

Basically where a notebook was empty it DIDN'T have the comment ```"# Databricks notebook source" ``` added to it and so the original attempt to remove it would return a bool with the value ```$false``` instead of an array minus the line ```"# Databricks notebook source" ```. 

I've got some more manual testing but hopefully the change will be added soon. I've also managed to fix the issue around the actual error not appearing.
